.github/workflows/tf_apply_production.yml

```diff
@@ -0,0 +1,50 @@
+name: "Terraform apply production"
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - ".github/manifests/infrastructure_version"
+
+env:
+  TERRAFORM_VERSION: 1.3.7
+  TERRAGRUNT_VERSION: 0.40.2
+  AWS_REGION: ca-central-1
+
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+  actions: write
+  checks: write
+  statuses: write
+
+jobs:
+  terragrunt-apply:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Get infrastructure version
+        run: echo "INFRASTRUCTURE_VERSION=$(cat .github/manifests/infrastructure_version)" >> $GITHUB_ENV
+
+      - name: Checkout infrastructure version tag
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          ref: ${{ env.INFRASTRUCTURE_VERSION }}
+
+      - name: Setup terraform tools
+        uses: cds-snc/terraform-tools-setup@v1
+
+      - name: Configure aws credentials using OIDC
+        uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # tag=v1.7.0
+        with:
+          role-to-assume: arn:aws:iam::806721586252:role/OIDCGithubWorkflowRole 
```

Comment:
This made me realize that I need to add the new OIDC roles we'll be using to manage these Terraform workflows.  I've got cds-snc/aft-account-request#96 open to do that.

Once it merges, we can update all these references to the new OIDC role names.

Reply:
So I see the new roles created - url-shortener-apply and url-shortener-plan.  I also created the OIDCGithubWorkflowRole in terragrunt/aws/api/oidc_iam.tf. Do I no longer need this file or do I need to update it?

Reply:
I suspect you won't need it anymore, but I'd need to see the file to be totally sure.  How about for this PR we just update the role names in the workflows and then we can sort out the what to do with `oidc_iam.tf` in a new PR?

Reply:
Sounds good! You just made me realize that I did not commit it for some reason and of course you did not see it. I can do another PR for that. Thank you for the detailed explanations and all your help!

Reply:
Done!